pkg/endpoint/metrics.go

```diff
@@ -74,3 +82,41 @@ func (s *regenerationStatistics) GetMap() map[string]time.Duration {
 		logfields.BuildDuration:  s.totalTime.Total(),
 	}
 }
+
+// endpointPolicyStatusMap is a map to store the endpoint id and the policy
+// enforcement status. Is used only for send metrics to prometheus.
+type endpointPolicyStatusMap struct {
+	sync.Map
+}
+
+// AddEndpoint add a new endpoint to the map and update the metrics related
+func (epPolicyMaps *endpointPolicyStatusMap) AddEndpoint(endpointID uint16, policyStatus models.EndpointPolicyEnabled) {
```

Comment:
This is not `Add()`. `Update()` is a better name here as the endpoint can already be tracked.